### PR TITLE
Add a definition for "frame"

### LINKED ISSUE #4266: Clarify frames and packets in Section 1
Section 1 is the first place that you introduce the term "Frame". I got very
confused about frames and packets, and I thing the taxonomy needs to be clearly
explained. You have UDP/IP packets that carry one or more QUIC PDUs, but the
frame/packet notation made it unclear what was being discussed and whether the
terms were being used consistently. Some clarification would be useful, or if
it is in the text then making it more obvious would help.

### PR DESCRIPTION
**Pull request by martinthomson**
I struggled with this a little.  Happy to take suggestions.

Closes #4266.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -239,6 +228,23 @@ Server:
 
 : The endpoint that accepts a QUIC connection.
 
+QUIC packet:
+
+: A complete processable unit of QUIC that can be encapsulated in a UDP
+  datagram.  QUIC packets carry one or more frames.  Multiple QUIC packets can
+  be encapsulated in a single UDP datagram.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:235]:
```suggestion
QUIC packet:

: A complete processable unit of QUIC that can be encapsulated in a UDP
  datagram.  One or more QUIC packets can be encapsulated in a single UDP
  datagram.
```

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -239,6 +228,23 @@ Server:
...
+Ack-eliciting Packet:
+
+: A QUIC packet that contains frames other than ACK, PADDING, and
+  CONNECTION_CLOSE. These cause a recipient to send an acknowledgment; see
+  {{sending-acknowledgements}}.
+
+Frame:
+
+: A unit of structured protocol information.  There are multiple frame types,
+  each of which carries different information.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:246]:
```suggestion
  each of which carries different information.  Collections of frames are
  transported by QUIC packets.
```

**Review by MikeBishop — COMMENTED** [review]
Being nitpicky, but...  Not all QUIC packets contain frames.  All frames are carried by packets, so let's define the relationship in that direction.

Also, I wonder if we should attempt to disambiguate QUIC packet versus IP packet; something like:

> Protocol elements called "packets" also exist in IP. Where IP packets are referenced, the protocol name will be stated explicitly, e.g. "IP packets."  References without a preface refer to QUIC packets as defined in this document.

...either in the definition or following the list.  WDYT?

**Comment by martinthomson** [comment]
I'm going to say that the IP Packet vs. QUIC Packet text is not necessary.  We carefully qualify where there is any potential for confusion.

**Review by janaiyengar — APPROVED** [review]
I like this definition. I'm happy with it.

**Comment by janaiyengar** [comment]
On the other issue, it might be worth simply stating the terminology upfront. Something like
```
We use the terms "QUIC packets", "UDP datagrams", and "IP packets" to refer to the
units of each protocol. That is, one or more QUIC packets can be encapsulated in a UDP
datagram, which in turn is encapsulated in an IP packet.
```

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -239,6 +228,24 @@ Server:
...
+Ack-eliciting Packet:
+
+: A QUIC packet that contains frames other than ACK, PADDING, and
+  CONNECTION_CLOSE. These cause a recipient to send an acknowledgment; see
+  {{sending-acknowledgements}}.
+
+Frame:
+
+: A unit of structured protocol information.  There are multiple frame types,
+  each of which carries different information.  Frames are conveyed in QUIC
```
Comment (ianswett) [inline draft-ietf-quic-transport.md:246]:
```suggestion
: A unit of structured protocol information.  A frame begins with a frame type,
  which indicates what information is contained in the frame. Frames are conveyed in QUIC
```

**Review by ianswett — APPROVED** [review]
One small suggestion, which you can ignore if you don't like it.

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (martinthomson) [reply draft-ietf-quic-transport.md:246]: I think we can skip that.  It's correct, but I don't think we need it here.

Comment (MikeBishop) [reply draft-ietf-quic-transport.md:246]: I agree; that's too detailed for the definitions list.  I think the furthest I'd go is to say there are multiple types of frames, but I'm not sure we need even that.

Comment (janaiyengar) [inline draft-ietf-quic-transport.md:246]:
```suggestion
  each of which carries different information.  Frames are contained in QUIC
```

Comment (martinthomson) [inline draft-ietf-quic-transport.md:246]:
```suggestion
  each of which carries different information.  Frames are contained in QUIC
```

**Pull request merged**